src/crimson/os/seastore/async_cleaner.h

```diff
@@ -832,8 +924,17 @@ class SegmentCleaner : public SegmentProvider {
     return space_tracker->equals(tracker);
   }
 
-  void set_disable_trim(bool val) {
-    disable_trim = val;
+  void set_journal(Journal *j) {
```

Comment:
Previously, the cleaner didn't need direct access to Journal because the relevant journal state and dirty extents are accessible via other pathways.  What specifically does the cleaner need access to now for a CBJounral that it doesn't already have via other means?  If we really do need the cleaner to have direct access to the Journal, can we clean up some of the other notification mechanisms?

Reply:
Removed, and add get_journal() in ExtentCallbackInterface (see [ae4fcd5](https://github.com/ceph/ceph/pull/46725/commits/ae4fcd53b7ec8864859c5544f6c17db052663e9c)). I think that we need journal information here to get the journal type and to call update_journal_tail after trimming as I explained  https://github.com/ceph/ceph/pull/46725#discussion_r910830759.